lasagne/layers/helper.py

```diff
@@ -183,6 +189,13 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
                                  "mapping this layer to an input expression."
                                  % layer)
             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
+            if not isinstance(layer, Mock) and \
+                    not isinstance(layer.get_output_for, Mock):
+                unused_kwargs -= set(getargspec(layer.get_output_for)[0])
```

Comment:
It might be good to just collect the actual keyword arguments, i.e., the ones that have default values set:
```
names, _, _, defaults = getargspec(layer.get_output_for)
unused_kwargs -= set(names[-len(defaults):])
```